tests/test_geoserver.py

```diff
@@ -154,28 +227,513 @@ def test_duplicate_datastore(self, geoserver):
         assert "Operation [_create_datastore_request] failed" in str(error.value)
 
     def test_publish_and_remove_shapefile(self, geoserver):
-        # Creating workspace and datastore needed to publish a shapefile
-        workspace_name = self.workspaces["publish_remove"]
-        datastore_name = "test-datastore-publish_remove"
-        geoserver_workspace_path = f"/user_workspaces/{workspace_name}/shapefile_datastore"
-        geoserver._create_workspace_request(workspace_name=workspace_name)
-        geoserver._create_datastore_request(workspace_name=workspace_name,
-                                            datastore_name=datastore_name)
-        geoserver._configure_datastore_request(workspace_name=workspace_name,
-                                               datastore_name=datastore_name,
-                                               datastore_path=geoserver_workspace_path)
-
-        # Preparations needed to make tests work without all the other handlers running
-        shapefile_name = "Espace_Vert"
-        workspace_path = self.folders["publish_remove"] + "/shapefile_datastore"
-        # This next part can fail if the user running this test doesn't have write access to the directory
-        copy_shapefile(basename=shapefile_name, destination=workspace_path)
+        workspace_name, datastore_name = prepare_geoserver_test_workspace(self, geoserver, "publish_remove")
 
         # Validate and publish shapefile
-        geoserver.validate_shapefile(workspace_name=workspace_name, shapefile_name=shapefile_name)
-        response = geoserver._publish_shapefile_request(workspace_name, datastore_name, shapefile_name)
+        geoserver.validate_shapefile(workspace_name=workspace_name, shapefile_name=self.test_shapefile_name)
+        response = geoserver._publish_shapefile_request(workspace_name, datastore_name, self.test_shapefile_name)
         assert response.status_code == 201
 
         # Remove shapefile
-        response = geoserver._remove_shapefile_request(workspace_name, datastore_name, shapefile_name)
+        response = geoserver._remove_shapefile_request(workspace_name, datastore_name, self.test_shapefile_name)
         assert response.status_code == 200
+
+
+# pylint: disable=W0201
+@pytest.mark.geoserver
+@pytest.mark.magpie
```

Comment:
Also `online`? Based on the `"http://localhost:9000/geoserver"` reference in the setup function.